### PR TITLE
feature: upgrade provider to work with gopass 1.10

### PR DESCRIPTION
**Pull request by jaceq**
This is a copy / update /followup of: https://github.com/camptocamp/terraform-provider-pass/pull/43

@raphink  I've made a change you requested last in original PR, I tested this locally and it works.
Please review and let me know if there are any outstanding issues.

Big thanks to @toabi for original PR.

### CODE REVIEW

File: pass/provider.go [hunk 1]
```diff
@@ -2,81 +2,40 @@ package pass
...
+func Provider() *schema.Provider {
 	return &schema.Provider{
-		Schema: map[string]*schema.Schema{
-			"store_dir": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				DefaultFunc: schema.EnvDefaultFunc("PASSWORD_STORE_DIR", ""),
-				Description: "Password storage directory to use.",
-			},
-			"refresh_store": {
```
Comment (raphink) [inline pass/provider.go:34]: Is the new behavior to never refresh, or to refresh always?

Comment (jaceq) [reply pass/provider.go:34]: Idea is to never refresh.

It was set so by original author, it came from this ticket: https://github.com/camptocamp/terraform-provider-pass/issues/42 , refresh is broken if local store has no remote.

Comment (toabi) [reply pass/provider.go:34]: Well the new public API just doesn't allow refresh (anymore).

Comment (raphink) [reply pass/provider.go:34]: Right, it seems the new API doesn't let you access Git at all (which is problematic imo).

**Comment by raphink** [comment]
Since the original work was done by @toabi , it would be good to list him [with a `Co-authored-by:` field](https://docs.github.com/en/free-pro-team@latest/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors)

**Comment by jaceq** [comment]
@raphink Absolutely, I should've done that in a first place. Added as requested.

btw. I see you are in Lausanne, greetings from Zurich ;)

**Pull request merged**